python/torch_mlir_e2e_test/test_suite/index_put.py

```diff
@@ -818,3 +818,28 @@ def forward(self, input, index, value):
 def IndexPutHackedTwin3DIntAccumulateModule_basic(module, tu: TestUtils):
     module.forward(tu.randint(10, 8, 6, high=1000), tu.randint(5, high=4),
                    tu.randint(5, 8, 6, high=1000))
+
+
+class IndexPutImpl2DIndexModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([4, 7], torch.float32, True),
+        ([1, 3], torch.int64, True),
+        ([1, 3, 7], torch.float32, True),
```

Comment:
If you use the less restrictive approach, you should no longer need to have static shapes here